libs/sdk/src/lib/kinetic-sdk-internal.ts

```diff
@@ -221,28 +223,57 @@ export class KineticSdkInternal {
     const destination = options.destination.toString()
     const senderCreate = options.senderCreate || false
 
-    this.validateDestination(appConfig, destination)
+    // We get the token account for the owner
+    const ownerTokenAccount = await this.findTokenAccount({
+      account: options.owner.publicKey,
+      commitment,
+      mint: mint.publicKey,
+    })
+
+    // The operation fails if the owner doesn't have a token account for this mint
+    if (!ownerTokenAccount) {
+      throw new Error(`Owner account doesn't exist for mint ${mint.publicKey}.`)
+    }
+
+    // We get the account info for the destination
+    const destinationTokenAccount = await this.findTokenAccount({
+      account: destination,
+      commitment,
+      mint: mint.publicKey,
+    })
+
+    // The operation fails if the destination doesn't have a token account for this mint and senderCreate is not set
+    if (!destinationTokenAccount && !senderCreate) {
+      throw new Error(`Destination account doesn't exist for mint ${mint.publicKey}.`)
+    }
 
-    const accounts = await this.getTokenAccounts({ account: destination, mint: mint.publicKey })
+    // Derive the associated token address if the destination doesn't have a token account for this mint and senderCreate is set
+    let senderCreateTokenAccount: PublicKeyString | undefined
+    if (!destinationTokenAccount && senderCreate) {
+      senderCreateTokenAccount = await getTokenAddress({ account: destination, mint: mint.publicKey })
+    }
 
-    if (!accounts?.length && !senderCreate) {
-      throw new Error(`Destination account doesn't exist.`)
+    // The operation fails if there is still no destination token account
+    if (!destinationTokenAccount && !senderCreateTokenAccount) {
+      throw new Error('Destination token account not found.')
     }
```

Comment:
three if statements in a row that start with `!destinationTokenAccount &&` should probably be refactored